Client/Frontend/Browser/TabManager.swift

```diff
@@ -414,7 +414,18 @@ class TabManager: NSObject {
     /// - Parameter notify: if set to true, the delegate is called when a tab is
     ///   removed.
     func removeAllPrivateTabsAndNotify(_ notify: Bool) {
-        privateTabs.forEach({ removeTab($0, flushToDisk: true, notify: notify) })
+        // if there is a selected tab, it needs to be closed last
+        // this is important for TopTabs as otherwise the selection of the new tab
+        // causes problems as it may no longer be present.
+        // without this we get a nasty crash
+        guard let selectedTab = selectedTab,
```

Comment:
Can't we avoid all the checks here? 
If selectTab doesnt exist then the filter at the bottom will just return all the private tabs. Thats fine, we still need to remove them. 
If selectTab isnt in privateTabs thats also okay. We still need to remove all the private tabs.